app/controllers/request_controller.rb

```diff
@@ -167,6 +167,12 @@ def similar
   end
 
   def list
+    # respond with a 404 and do not execute the search if request was not for html
+    if request.format && !request.format.html?
+      respond_to { |format| format.any { head :not_found } }
+      return
+    end
```

Comment:
This has the advantage of returning from the controller early, without any potentially expensive hits on the db, otherwise all the controller code is run then not used for anything.